platform-sdk/docs/proposals/template.md

```diff
@@ -0,0 +1,109 @@
+# Summary
+
+Give a 1-3 sentence summary of the capability in the design proposal.
+
+| Metadata           | Entities                            | 
+|--------------------|-------------------------------------|
+| Last Updated       | 2024-06-07  (Edward)                |
+| Designers          | John, Mary, alpha-team              |
+| Functional Impacts | Services, DevOps, Mirror Node, etc. |
+| Related Proposals  | Proposal-1, Proposal-2              |
+| HIPS               | HIP-1, HIP-2,                       |
+
+# Purpose and Context
+
+Why is the design proposal necessary? Describe the background and purpose of the design proposal at a high level.
+
+## Dependencies, Interactions, and Implications
+
+What are the dependencies and pre-requisites of this proposal?  
+What other proposals, capabilities, and teams are impacted by this proposal?
+What does the proposal enable once it is completed?
+
+Remove this section if not applicable.
+
+# Requirements
+
+Describe the requirements and acceptance criteria that the design proposal must satisfy.
+
+# Design Decisions
+
+Describe the high level options available, the trade-offs between them and the decisions made.
+
+Remove this section if not applicable.
+
+# Changes
+
+## Public API
+
+Describe any public API changes or additions. Include stakeholders of the API.
+
+Code can be included in the proposal directory, but not submitted to the code base.
+
+Remove this section if not applicable.
+
+## Configuration
+
+Describe any new or modified configuration.
+
+Remove this section if not applicable.
+
+## Core Behaviors
+
+Describe any new or modified behavior. What are the new or modified algorithms and protocols? Include any diagrams that
+help explain the behavior.
+
+Remove this section if not applicable.
+
+## Components and Architecture
+
+Describe any new or modified components or architectural changes. This includes thread management changes, state
+changes, disk I/O changes, platform wiring changes, etc. Include diagrams of architecture changes.
+
+Remove this section if not applicable.
+
+## Metrics
+
+Are there new metrics? Are the computation of existing metrics changing? Are there expected observable metric impacts
+that change how someone should relate to the metric?
+
+Remove this section if not applicable.
+
+## Performance
+
+Describe any expected performance impacts. This section is mandatory for platform wiring changes.
+
+Remove this section if not applicable.
+
+## Modules and Repos
+
+Describe any new or modified modules or repositories.
+
+Remove this section if not applicable.
+
+# Test Plan
+
+## Unit Tests
+
+Describe the core behaviors that need new unit tests and any special considerations in test design, such as subtle edge
```

Comment:
I don't think unit tests, in the traditional sense of the term where only a single method is tested per test, is necessary here. What is more useful, IMO, is describing any higher level functionality that can be covered within the unit test framework. For example, there are various simulators in unit tests that test much more than a single function and are fairly complex, like the consensus unit test framework and the gossip and intake pipeline simulators. These sorts of things need to be planned out and included here.

Reply:
perhaps this is what is meant by `complex component interactions`?

Reply:
yes that is what I meant by complex component interactions.

Reply:
I've updated the wording.  I still think that the designers should call out any subtle behavior that someone else might be unaware of that needs to be tested.  Let me know if the tweaked wording is acceptable.